modules/core/src/main/java/org/locationtech/jts/precision/GeometryPrecisionReducer.java

```diff
@@ -173,8 +173,8 @@ private Geometry fixPolygonalTopology(Geometry geom)
   	
   	Geometry finalGeom = bufGeom;
   	if (! changePrecisionModel) {
-  	  // a slick way to copy the geometry with the original precision factory
-  		finalGeom = geom.getFactory().createGeometry(bufGeom);
+  		finalGeom = bufGeom.copy();
```

Comment:
Don't think this copy is needed - the GeometryEditor will copy.

Reply:
Hmm ok, I thought GeometryEditor with NoOpGeometryOperation did not.

Reply:
Regardless, I see now I get two failing tests so will have to rethink this.

Reply:
yes, I think you are right - using NoOpGeometryOperation does not copy the coordinates (but does copy the Geometry objects using the new factory holding the new PM).

Too bad - seems like extra work to copy the entire geometry just to copy the coordinates.
Sigh... GeometryEditor is pretty complicated, and easy to get wrong.

Reply:
Indeed. Would it make sense to have a setter for PrecisionModel on the Geometry to use instead?

Reply:
Or rather, set the PrecisionModel on the GeometryFactory.. but I tried that and the failures are still there, and reading you right this is what the NoOpGeometryOperation does too. So I'm confused about what the original code actually does now, it seems it does more than change the PrecisionModel.

Reply:
GeometryFactory must be immutable, so cannot have setter for PM.  Can just make a new factory if needed.  This is what the original code did. 

What is the problem with the original code?

Reply:
I get into problems with the transpilation of the anonymous GeometryEditor.CoordinateSequenceOperation, here https://github.com/locationtech/jts/blob/6b5829735c7b03b283f910cd88257da950d8f8d9/modules/core/src/main/java/org/locationtech/jts/geom/GeometryFactory.java#L607. The overload resolution comes out wrong for unknown reasons as of yet.

Other than that, I fail to understand what the original code does more than the comment `a slick way to copy the geometry with the original precision factory` appear to indicate.

Reply:
(if it did just copy the geometry using the original factory, I don't see why the alternative code would not work)

Reply:
The original code pretty much does what the comment says. But note that the createGeometry does not copy the coordinate objects, but this is ok in this context because they have already been copied by the buffer(0) operation.